### PR TITLE
When --out/output and outputter are used, --out/output will be used

### PR DESCRIPTION
**Pull request by ckochenower**
This fix for #20634 has been code reviewed by @cachedout

### CODE REVIEW

**Comment by terminalmage** [comment]
The remaining lint error is unrelated. It can be fixed elsewhere.

**Comment by terminalmage** [comment]
I have two pull requests that depend on this being merged. They will both need to be rebased and tested once this is merged. They are:

* https://github.com/saltstack/salt/pull/21320
* https://github.com/saltstack/salt/pull/21463

File: salt/client/mixins.py [hunk 1]
```diff
@@ -429,13 +440,29 @@ def print_async_event(self, suffix, event):
...
+        outputter = event.get('outputter', None)
+        # --out and --output are gracked internally with 'output'
+        # self.opts.get('out') will resolve to None
+        opts_outputter = self.opts.get('output', None)
+
+        # Override with opts_outputter?
+        if opts_outputter is not None:
+            # Emit a warning if both are defined and the outputter is changing
+            if outputter is not None and outputter != opts_outputter:
+                warning = "Both outputter=%s and --out/output=%s are defined. %s will be used." % (outputter, opts_outputter, opts_outputter)
```
Comment (kiorky) [inline salt/client/mixins.py]: %s; .format() ?

Comment (terminalmage) [inline salt/client/mixins.py]: Good call. @ckochenower, can you replace this with string formatting using .format()?

**Comment by terminalmage** [comment]
These test failures are legitimate. The state.over runner is only returning a single line of output

```
progress: {'message': 'Stage execution results:'}
```

This may just be a matter of the overstate code needing to still be modified to return data in the proper format. I will continue to investigate.

File: salt/client/mixins.py (hunk 1 above)
Comment (ckochenower) [inline salt/client/mixins.py]: Done. Still learning what coding conventions are imposed by the linter. Thanks for being patient with me.

**Comment by terminalmage** [comment]
I have found the cause of the test failures, and submitted a pull request against Corin's branch: https://github.com/ckochenower/salt/pull/1

**Pull request merged**